### PR TITLE
[MODLISTS-152] Migrate cross-tenant lists to private

### PR DESCRIPTION
**Pull request by ncovercash**

### CODE REVIEW

File: src/main/java/org/folio/list/services/CustomTenantService.java [hunk 1]
```diff
@@ -50,14 +50,13 @@ protected void afterTenantUpdate(TenantAttributes tenantAttributes) {
...
       .build()
       .execute(ctx -> {
-        log.info("Verifying lists are up to date. Attempt #" + (ctx.getRetryCount() + 1));
-        migrationService.verifyListsAreUpToDate();
+        log.info("Performing tenant install migrations. Attempt #" + (ctx.getRetryCount() + 1));
+        migrationService.performTenantInstallMigrations();
         return null;
       }, ctx -> {
-        log.error("Unable to verify lists are up to date", ctx.getLastThrowable());
-        // This RetryTemplate only deals with InsufficientEntityTypePermissionsException, so we know that's what this
```
Comment (ncovercash) [inline src/main/java/org/folio/list/services/CustomTenantService.java:58]: We can't, though — `getLastThrowable` can be _any_ exception thrown from `performTenantInstallMigrations`

**Review by kjain110 — APPROVED** [review]

File: src/main/java/org/folio/list/services/CustomTenantService.java (hunk 1 above)
Comment (mweaver-ebsco) [reply src/main/java/org/folio/list/services/CustomTenantService.java:58]: Ugh, you're right! I must have done something stupid when I tested that >_<

**Review by bvsharp — APPROVED** [review]

File: src/main/java/org/folio/list/domain/ListEntity.java [hunk 2]
```diff
@@ -31,7 +33,7 @@
 @AllArgsConstructor
 @NoArgsConstructor
 @Table(name = "list_details")
-public class ListEntity {
+public class ListEntity implements Serializable {
```
Comment (mweaver-ebsco) [inline src/main/java/org/folio/list/domain/ListEntity.java:36]: Do we actually need these classes to be `Serializable`? `Serializable` has some warts and should generally be avoided unless you actually _need_ it

Comment (ncovercash) [reply src/main/java/org/folio/list/domain/ListEntity.java:36]: Side effect from it getting stored as a field in our exceptions, sonar wanted it serializable. I don't see any reason these entities can't be, though?

Comment (mweaver-ebsco) [reply src/main/java/org/folio/list/domain/ListEntity.java:36]: I always forget most of the details, but _Effective Java_ has an entire chapter about `Serializable` just pretty much boils down to, "don't (unless you really, really, really, _really_, need it; in which case, be very, very, very careful")

**Review by mweaver-ebsco — APPROVED** [review]

**Pull request merged**